### ISSUE TITLE
[29-02-2020]: [Markdown processor isn't "smart"]

### ISSUE THREAD

**Issue by aadibajpai** [fastai/fastpages#142]
## Prerequisites for filing a bug

- [ ] Did you run the local development server as indicated in the [development guide](https://github.com/fastai/fastpages/blob/master/DEVELOPMENT.md)?  
  - Please include any useful logs or error messages from that
- [x] Have you searched opened / closed issues on this repo for a similar problem?
- [ ] Have you searched the [fastai forums](https://forums.fast.ai/) for a similar problem?

## Please include the following information

1. Steps to reproduce the problem
Create a hyperlink in your markdown post containing `[]` or `|` in the text.
2. A link to your repository where error is occuring
Private repository for now but the relevant post can be seen here https://aadibajpai.github.io/ccx.org/2019/09/03/blog-posts-our-students.html
Look at the link rendering errors.
3. A link to the notebook or markdown file where the error is occuring
https://pastebin.com/YFgfruYz
4. If the error is happening in GitHub Actions, a link to the specific error along with how you are able to reproduce this error.
5. A screenshot / dump of relevant logs or error messages you are receiving from your local development environment.

When I preview the markdown file in Sublime, its syntax highlighting is smart enough to parse the hyperlinks but I assume fastpages renderer has a problem—the | gets interpreted as tables and the [] interfere with the [] from the hyperlink syntax.

![image](https://user-images.githubusercontent.com/27063113/75597261-80d3c300-5aba-11ea-91c4-dba7581b176b.png)

Expected behaviour would be that tags are matched.

**Comment by hamelsmu** [comment]
You need to escape the vertical bars with `\`.  Example

`week 1 \| GSoc’19 \| CCExtractor Development`

**Comment by hamelsmu** [comment]
@aadibajpai Also I noticed that your front matter is incorrect.  I also noticed that my example front matter is incorrect in the README which I corrected via https://github.com/fastai/fastpages/pull/147

So instead of 

```yaml
---
- title = "Blog Posts from our Students"
- toc: true
---
```

It should be 

```yaml
---
title : "Blog Posts from our Students"
toc: true
---
```

**Comment by aadibajpai** [comment]
> @aadibajpai Also I noticed that your front matter is incorrect. I also noticed that my example front matter is incorrect in the README which I corrected via #147
> 
> So instead of
> 
> ```yaml
> ---
> - title = "Blog Posts from our Students"
> - toc: true
> ---
> ```
> 
> It should be
> 
> ```yaml
> ---
> title: "Blog Posts from our Students"
> toc: true
> ---
> ```

Yeah, I had ended up doing this manually via the online editor. Would have opened a PR haha but thanks for fixing that! 👍
